src/components/AudioPlayer/SeekButton.tsx

```diff
@@ -1,30 +1,76 @@
 import Button, { ButtonShape, ButtonVariant } from 'src/components/dls/Button/Button';
-import { withStopPropagation } from 'src/utils/event';
+import { selectAudioData, selectReciter } from 'src/redux/slices/AudioPlayer/state';
+import { selectHighlightedLocation } from 'src/redux/slices/QuranReader/highlightedLocation';
+import { makeChapterAudioDataUrl } from 'src/utils/apiPaths';
+import { getVerseTimingByVerseKey } from 'src/utils/audio';
+import { getChapterData } from 'src/utils/chapter';
+import { makeVerseKey } from 'src/utils/verse';
 
 export enum SeekButtonType {
-  FastForward = 'fastForward',
-  Rewind = 'rewind',
+  NextAyah = 'nextAyah',
+  PrevAyah = 'prevAyah',
 }
 
 type SeekButtonProps = {
   type: SeekButtonType;
   isLoading: boolean;
 };
 const SeekButton = ({ type, isLoading }: SeekButtonProps) => {
+  const { highlightedChapter, highlightedVerse } = useSelector(selectHighlightedLocation);
+  const reciter = useSelector(selectReciter);
+  const audioData = useSelector(selectAudioData);
+  const chapterData = useMemo(
+    () => getChapterData(highlightedChapter?.toString()),
+    [highlightedChapter],
+  );
+
+  const { data: chapterAudioData } = useSWRImmutable(
+    reciter.id && audioData?.chapterId
+      ? makeChapterAudioDataUrl(reciter.id, audioData?.chapterId, true)
+      : null, // only fetch when reciterId and chapterId is available
+    () => getChapterAudioData(reciter.id, audioData?.chapterId, true),
+  );
+
+  const verseTimingData = chapterAudioData?.verseTimings || [];
+
+  const onSeek = () => {
+    const verse = type === SeekButtonType.PrevAyah ? highlightedVerse - 1 : highlightedVerse + 1;
+    const verseKey = makeVerseKey(highlightedChapter, verse);
+
+    const selectedVerseTiming = getVerseTimingByVerseKey(verseKey, verseTimingData);
+    triggerSetCurrentTime(selectedVerseTiming.timestampFrom / 1000); // AudioPlayer accept 'seconds' instead of 'ms'
+  };
+
+  // disable the button if loading, or chapterAudioData not available, or highlighted verse not available
+  // or currently playing first verse or last verse
+  const isDisabled =
+    isLoading ||
+    !highlightedChapter ||
+    !highlightedVerse ||
+    !chapterAudioData ||
+    (type === SeekButtonType.PrevAyah && highlightedVerse <= 1) ||
+    (type === SeekButtonType.NextAyah && highlightedVerse >= chapterData?.versesCount);
+
   return (
     <Button
-      tooltip={type === SeekButtonType.Rewind ? 'Rewind 10 seconds' : 'Fast forward 10 seconds'}
+      tooltip={type === SeekButtonType.PrevAyah ? 'Previous Ayah' : 'Next Ayah'}
       variant={ButtonVariant.Ghost}
       shape={ButtonShape.Circle}
-      disabled={isLoading}
-      onClick={withStopPropagation(() => triggerSeek(type === SeekButtonType.Rewind ? -10 : 10))}
+      disabled={isDisabled}
+      onClick={onSeek}
     >
-      {type === SeekButtonType.Rewind ? <ReplayIcon /> : <ForwardIcon />}
+      {type === SeekButtonType.PrevAyah ? <PrevIcon /> : <NextIcon />}
```

Comment:
2 comments on this: 

1. Can we use this icons instead? I believe the ones you used indicate playing the previous/next track
![image](https://user-images.githubusercontent.com/11590314/136653192-6783c968-0af9-4536-ac1c-d381b2e97970.png)

2. Can we use icons that use the same size as the other audio player icons? the ones on the deployment appear a bit bigger